lib/node/runner/node_runner.go

```diff
@@ -659,12 +659,12 @@ func (nr *NodeRunner) proposeNewBallot(round uint64) (ballot.Ballot, error) {
 	}
 
 	// remove invalid transactions
-	if len(transactionsChecker.InvalidTransactions()) > 0 {
-		nr.TransactionPool.Remove(transactionsChecker.InvalidTransactions()...)
+	if len(transactionsChecker.invalidTransactions()) > 0 {
```

Comment:
Instead of iterating through the transactions 3 times, why don't we store the result ?

Reply:
Good idea :) Please check 58439cc60e08d1f34f668ae3f722a30986f09000